rmf_traffic/src/rmf_traffic/Spline.cpp

```diff
@@ -67,18 +67,18 @@ Spline::Parameters compute_parameters(
   const Trajectory::const_iterator start_it =
       --Trajectory::const_iterator(finish_it);
 
-  const Trajectory::Segment& start = *start_it;
-  const Trajectory::Segment& finish = *finish_it;
+  const Trajectory::Waypoint& start = *start_it;
+  const Trajectory::Waypoint& finish = *finish_it;
 
-  const Time start_time = start.get_finish_time();
-  const Time finish_time = finish.get_finish_time();
+  const Time start_time = start.time();
+  const Time finish_time = finish.time();
 
   const double delta_t = compute_delta_t(finish_time, start_time);
 
-  const Eigen::Vector3d x0 = start.get_finish_position();
-  const Eigen::Vector3d x1 = finish.get_finish_position();
-  const Eigen::Vector3d v0 = delta_t * start.get_finish_velocity();
-  const Eigen::Vector3d v1 = delta_t * finish.get_finish_velocity();
+  const Eigen::Vector3d x0 = start.position();
+  const Eigen::Vector3d x1 = finish.position();
+  const Eigen::Vector3d v0 = delta_t * start.velocity();
+  const Eigen::Vector3d v1 = delta_t * finish.velocity();
 
   const rmf_traffic::Trajectory::ConstProfilePtr profile_ptr =
```

Comment:
This was added in PR #68 to access the `characteristic_length` of the profile's shape when computing the bounding box for braodpahse collision detection. As the trajectory iterators no longer contain profiles, this would result in an error. I guess `Spline::Parameters` should no longer contain the `profile_ptr` element. We will need to modify `DetectConflict::broad_phase` suitably.

Reply:
Yep, I'm doing a redesign/refactor of `Collision.hpp` right now.